client/src/reducers/controls.ts

```diff
@@ -457,7 +457,7 @@ const Controls = (
     case "toggle image underlay": {
       return {
         ...state,
-        imageUnderlay: !state.imageUnderlay,
```

Comment:
let's check the existing state and only return a new state if it's different, so we don't unnecessarily trigger a re-render.

E.g.,

```ts
case "toggle image underlay": {
  if (state.imageUnderlay === action.toggle) return state;

  return { ... }
```